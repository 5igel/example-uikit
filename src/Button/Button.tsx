import React, { ButtonHTMLAttributes, MouseEventHandler } from 'react'
import styled from '@emotion/styled'
import { variant, padding, typography, border, PaddingProps, TypographyProps, BorderProps } from 'styled-system'
import shouldForwardProp from '@styled-system/should-forward-prop'


export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement>, PaddingProps, TypographyProps, BorderProps {
  label: string;
  onClick?: MouseEventHandler;
  variant?: 'primary' | 'secondary' | 'danger'
}

const BaseButton: React.FC<ButtonProps> = ({
  label,
  ...props
}) => {
  return (
    <button
      type="button"
      {...props}
    >
      {label}
    </button>
  );
};

export const Button = styled(BaseButton, {
  shouldForwardProp,
})(
  {
    appearance: 'none',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
  },
  padding,
  typography,
  border,
  variant({
    variants: {
      primary: {
        color: 'background',
        bg: 'primary',
      },
      secondary: {
        color: 'foreground',
        bg: 'secondary',
      },
      danger: {
        color: 'background',
        bg: 'danger',
      },
    }
  })
)

Button.defaultProps = {
  variant: 'primary',
  py: 2,
  px: 4,
  border: 0,
  borderRadius: 1,
  fontSize: 0,
  fontWeight: 2,
}
