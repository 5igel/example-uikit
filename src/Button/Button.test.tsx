import React from 'react';
import { render } from '@testing-library/react';

import {Button} from './Button';

test('renders button', () => {
  const { container, getByText } = render(<Button label="snapshot" />)
  expect(container.firstChild).toMatchSnapshot()
});

test('renders button with html attributes', () => {
  const { container, getByText } = render(<Button label="snapshot" type="submit" data-custom="custom" />)
  expect(container.firstChild).toMatchSnapshot()
});
