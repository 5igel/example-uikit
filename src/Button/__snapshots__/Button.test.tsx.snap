// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button 1`] = `
<button
  class="css-1iwkoki"
  type="button"
>
  snapshot
</button>
`;

exports[`renders button with html attributes 1`] = `
<button
  class="css-1iwkoki"
  data-custom="custom"
  type="submit"
>
  snapshot
</button>
`;
